src/containers/announcements/ducks/thunks.ts

```diff
@@ -1,19 +1,21 @@
 import { Announcement, AnnouncementsThunkAction } from './types';
 import { announcements } from './actions';
 
-export const getAnnouncements = (limit?: number): AnnouncementsThunkAction<void> => {
-    return (dispatch, getState, { publicApiClient }) => {
-        dispatch(announcements.loading());
-        return publicApiClient
-            .getAnnouncements()
-            .then((response: Announcement[]) => {
-                return (limit && limit >= 0) ? response.slice(0, limit) : response;
-            })
-            .then((response: Announcement[]) => {
-                dispatch(announcements.loaded(response));
-            })
-            .catch((error: any) => {
-                dispatch(announcements.failed(error));
-            });
-    };
+export const getAnnouncements = (
+  limit?: number,
+): AnnouncementsThunkAction<void> => {
+  return (dispatch, getState, { publicApiClient }) => {
+    dispatch(announcements.loading());
+    return publicApiClient
+      .getAnnouncements()
+      .then((response: Announcement[]) => {
+        return limit && limit >= 0 ? response.slice(0, limit) : response;
```

Comment:
If we're getting the whole response from the api, we should just keep it and store it, OR we can alter the api

Reply:
+1. Maybe we can slice it using a limit, if specified, when we're just about to render it, instead of when storing it. That way at least we have all the data